qtconsole/mainwindow.py

```diff
@@ -809,7 +810,16 @@ def set_syntax_style(self, syntax_style):
         self.active_frontend._syntax_style_changed()
         self.active_frontend._style_sheet_changed()
         self.active_frontend.reset(clear=True)
-        self.active_frontend._execute("%colors linux", True)
+        self.active_frontend._execute(
+f"""
+from IPython.core.ultratb import VerboseTB
+try:
+    VerboseTB._tb_highlight_style = '{syntax_style}'
+except AttributeError:
+    get_ipython().run_line_magic('colors', '{colors}')
```

Comment:
Following the response at https://github.com/ipython/ipython/issues/14464#issuecomment-2180056302 I think now this could be:

```suggestion
from IPython.core.ultratb import VerboseTB
if getattr(VerboseTB, 'tb_highlight_style', None) is not None:
    VerboseTB.tb_highlight_style = '{syntax_style}'
elif getattr(VerboseTB, '_tb_highlight_style', None) is not None:
    VerboseTB._tb_highlight_style = '{syntax_style}'
else:
    get_ipython().run_line_magic('colors', '{colors}')
```

Reply:
Yep, that should do the trick 👍🏽